internal/clients/grpc/connect.go

```diff
@@ -45,20 +50,42 @@ func NewClientHandler(c *Client) ClientHandler {
 
 // Connect creates new gRPC client and connect to gRPC server.
 func Connect(cfg Config) (*gogrpc.ClientConn, bool, error) {
-	var opts []gogrpc.DialOption
+	opts := []gogrpc.DialOption{
+		gogrpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
+	}
 	secure := false
 	tc := insecure.NewCredentials()
 
-	if cfg.ClientTLS && cfg.CACerts != "" {
-		var err error
-		tc, err = credentials.NewClientTLSFromFile(cfg.CACerts, "")
+	if cfg.ServerCAFile != "" {
+		tlsConfig := &tls.Config{}
+
+		// Loading root ca certificates file
+		rootCA, err := ioutil.ReadFile(cfg.ServerCAFile)
 		if err != nil {
-			return nil, secure, err
+			return nil, secure, fmt.Errorf("failed to load root ca file: %w", err)
+		}
+		if rootCA != nil {
+			capool := x509.NewCertPool()
+			if !capool.AppendCertsFromPEM(rootCA) {
+				return nil, secure, fmt.Errorf("failed to append root ca to tls.Config")
+			}
+			tlsConfig.RootCAs = capool
 		}
+
+		// Loading mtls certificates file
+		if cfg.ClientCert != "" || cfg.ClientKey != "" {
```

Comment:
Since `tls.LoadX509KeyPair` will return an error if either is empty

```suggestion
		if cfg.ClientCert != "" && cfg.ClientKey != "" {
```

Reply:
@rodneyosodo  As said before , the same reason, 

Since there is no variable to check weather user is opted from HTTPS ,

If user fails to provide two file, Instead provided one file, Then program will run without TLS if we use  `&&`

Instead if we use `||` if one file is provided it will lead to error , So by user will get notified

Reply:
Okay. I see.